contracts/src/interfaces/IPayer.sol

```diff
@@ -0,0 +1,598 @@
+     * @return The current balance of the payer.
+     */
+    function getPayerBalance(address payer) external view returns (uint256);
+
+    /**
+     * @notice Initiates a withdrawal request for the caller.
+     *         - Sets the payer into withdrawal mode (no further usage allowed).
+     *         - Records a timestamp for the withdrawal lock period.
+     * @param amount The amount to withdraw (can be less than or equal to current balance).
+     *
+     * Emits `WithdrawRequested`.
+     */
+    function requestWithdraw(uint256 amount) external;
+
+    /**
+     * @notice Cancels a previously requested withdrawal, removing withdrawal mode.
+     * @dev Only callable by the payer who initiated the withdrawal.
+     *
+     * Emits `WithdrawCancelled`.
+     */
+    function cancelWithdraw() external;
+
+    /**
+     * @notice Finalizes a payer's withdrawal after the lock period has elapsed.
+     *         - Accounts for any pending usage during the lock.
+     *         - Returns the unspent balance to the payer.
+     *
+     * Emits `WithdrawFinalized`.
+     */
+    function finalizeWithdraw() external;
+
+    /**
+     * @notice Checks if a payer is currently in withdrawal mode and the timestamp
+     *         when they initiated the withdrawal.
+     * @param payer The address to check.
+     * @return inWithdrawal True if in withdrawal mode, false otherwise.
+     * @return requestTimestamp The timestamp when `requestWithdraw()` was called.
+     * @return withdrawableTimestamp When the withdrawal can be finalized.
+     * @return amount The amount requested for withdrawal.
+     */
+    function getWithdrawalStatus(address payer)
+        external
+        view
+        returns (bool inWithdrawal, uint256 requestTimestamp, uint256 withdrawableTimestamp, uint256 amount);
+
+    /**
+     * @notice Returns the duration of the lock period required before a withdrawal
+     *         can be finalized.
+     * @return The lock period in seconds.
+     */
+    function getLockPeriod() external view returns (uint256);
+
+    //==============================================================
+    //                       USAGE SETTLEMENT
+    //==============================================================
+
+    /**
+     * @notice Called by node operators to settle usage and calculate fees owed.
+     * @dev This function is EIP-1283 optimized by using accumulators for multiple state updates.
+     * @param fees The total USDC fees computed from this usage period.
+     * @param payer The address of the payer being charged.
+     * @param nodeId The ID of the node operator submitting the usage.
+     * @param messageCount Number of messages processed during this period.
+     * @param dataSize Total size of data processed in bytes.
+     * @param timestamp The timestamp when the usage occurred (can be backdated).
+     * @param commitmentHash Optional hash for additional off-chain data verification.
```

Comment:
What would the `commitmentHash` typically be set to?

Reply:
This is a typo, thanks for catching it. The commitmentHash should be calculated by the settleUsage itself and not passed as param (security risk).

As mentioned before, the hash can be hash(payer, fee, nodeId, timestamp/block), any combination that gives us unique hashes and be used for verification. It's optional given our current design and we can't live without it.

Reply:
It does seem like we want to somehow guarantee uniqueness of the report, so I'm in favour. But yes, let's have the smart contract handle the hashing.